src/blob/core.rs

```diff
@@ -538,8 +538,11 @@ impl RawRecords {
         debug!("blob raw records load");
         let mut headers = Vec::new();
         while self.current_offset < self.file.size() {
-            let header = self.read_current_record_header().await.with_context(|| {
-                format!("read record header failed, at {}", self.current_offset)
+            let (header, data) = self.read_current_record_header_and_data().await.with_context(|| {
```

Comment:
This should be configurable behavior: configuration parameter value should come from `node.yaml`. By default checksum validation should be disabled, but can be enabled through config